tests/e2e/slashing.go

```diff
@@ -288,9 +288,8 @@ func (suite *CCVTestSuite) TestOnRecvSlashPacketErrors() {
 	errAck := providerKeeper.OnRecvSlashPacket(ctx, packet, packetData)
 	suite.Require().False(errAck.Success())
 	errAckCast := errAck.(channeltypes.Acknowledgement)
-	suite.Require().Equal(
-		fmt.Sprintf("cannot find infraction height matching the validator update id 0 for chain %s",
-			firstBundle.Chain.ChainID), errAckCast.GetError())
+	// TODO: see if there's a way to get error reason like before
```

Comment:
This would be nice to fix before merging, but is not high prio imo and could be tracked by issue if needed

Reply:
Unfortunately the error string info is not included into the error ack as before. This was changed to prevent risks of app hash mismatches in the event that a dev changes an error string in `OnRecvPacket` which would be bubbled up and included into the error acknowledgement which is written to the ibc core store. 

Instead we parse out the abci code and emit a generic string. https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/types/acknowledgement.go#L32-L42

For `OnRecvPacket`, in the event of an error and returning an error ack, you can include error details in the events emitted. You could get at them in the test code, but may require a little bit of refactoring.

You'd need to build the `MsgRecvPacket` and route it via the testchain [`SendMsgs(...)`](https://github.com/cosmos/ibc-go/blob/main/testing/chain.go#L316) method. Something like:

```go
msgRecvPacket := &channeltypes.MsgRecvPacket{...testvalues}

res, err := suite.providerChain.SendMsgs(msgRecvPacket)
suite.Require().NoError(err)

events := res.GetEvents()
// make any assertions on events
```

Reply:
Thanks a lot for these instructions @damiannolan :raised_hands: As it necessitates the expansion of the test setup logic, mostly in order to execute SendMsg successfully, we will implement this in a separate pull request.

Reply:
Is there an issue regarding Damian's comment?

Reply:
This issue https://github.com/cosmos/interchain-security/issues/546 will require Damian's solution to be properly tested.